pkgs/native_assets_builder/lib/src/package_layout/package_layout.dart

```diff
@@ -29,26 +30,30 @@ class PackageLayout {
 
   final Uri packageConfigUri;
 
-  PackageLayout._(
-      this.rootPackageRoot, this.packageConfig, this.packageConfigUri);
+  PackageLayout._(this._fileSystem, this.rootPackageRoot, this.packageConfig,
+      this.packageConfigUri);
 
   factory PackageLayout.fromPackageConfig(
+    FileSystem fileSystem,
     PackageConfig packageConfig,
     Uri packageConfigUri,
   ) {
-    assert(File.fromUri(packageConfigUri).existsSync());
+    assert(fileSystem.file(packageConfigUri).existsSync());
     packageConfigUri = packageConfigUri.normalizePath();
     final rootPackageRoot = packageConfigUri.resolve('../');
-    return PackageLayout._(rootPackageRoot, packageConfig, packageConfigUri);
+    return PackageLayout._(
+        fileSystem, rootPackageRoot, packageConfig, packageConfigUri);
   }
 
-  static Future<PackageLayout> fromRootPackageRoot(Uri rootPackageRoot) async {
+  static Future<PackageLayout> fromRootPackageRoot(
+      FileSystem fileSystem, Uri rootPackageRoot) async {
```

Comment:
Nit: here and other places, please insert trailing comma to avoid reformatting when we turn the 3.7 formatter on. 🤓  (Otherwise our whole git blame disappears on 3.7 when we turn the formatter on.)

Reply:
Done.